app/gui2/src/stores/suggestionDatabase/entry.ts

```diff
@@ -0,0 +1,151 @@
+
+export interface SuggestionEntry {
+  kind: SuggestionKind
+  /// A module where the suggested object is defined.
+  definedIn: QualifiedName
+  /// A type or module this method or constructor belongs to.
+  memberOf?: QualifiedName
+  isPrivate: boolean
+  isUnstable: boolean
+  /// A name of suggested object.
+  name: string
+  /// A list of aliases.
+  aliases: string[]
+  /// A type of the "self" argument. This field is present only for instance methods.
+  selfType?: QualifiedName
+  /// Argument lists of suggested object (atom or function). If the object does not take any
+  /// arguments, the list is empty.
+  arguments: SuggestionEntryArgument[]
+  /// A type returned by the suggested object.
+  returnType: QualifiedName
+  /// A module reexporting this entity.
+  reexportedIn?: QualifiedName
+  /// A list of documentation sections associated with object.
+  documentation: string
+  /// A scope where this suggestion is visible.
+  scope?: SuggestionEntryScope
+  /// A name of a custom icon to use when displaying the entry.
+  iconName?: string
+  /// A name of a group this entry belongs to.
+  groupIndex?: number
+}
+
+export function makeSimpleEntry(
+  kind: SuggestionKind,
+  definedIn: QualifiedName,
+  name: string,
+  returnType: QualifiedName,
+): SuggestionEntry {
+  return {
+    kind,
+    definedIn,
+    name,
+    isPrivate: false,
+    isUnstable: false,
+    aliases: [],
+    arguments: [],
+    returnType,
+    documentation: '',
+  }
+}
+
+export function makeModule(definedIn: QualifiedName) {
+  return makeSimpleEntry(SuggestionKind.Module, definedIn, qnLastSegment(definedIn), definedIn)
+}
+
+export function makeType(definedIn: QualifiedName, name: string) {
+  return makeSimpleEntry(SuggestionKind.Type, definedIn, name, `${definedIn}.${name}`)
+}
+
+export function makeCon(type: QualifiedName, name: string) {
+  return {
+    memberOf: type,
+    ...makeSimpleEntry(SuggestionKind.Constructor, qnParent(type), name, type),
+  }
+}
+
+export function makeMethod(type: QualifiedName, name: string, returnType: string) {
+  return {
+    memberOf: type,
+    selfType: type,
+    ...makeSimpleEntry(SuggestionKind.Method, qnParent(type), name, returnType),
+  }
+}
+
+export function makeStaticMethod(type: QualifiedName, name: string, returnType: string) {
+  return {
+    memberOf: type,
+    ...makeSimpleEntry(SuggestionKind.Method, qnParent(type), name, returnType),
+  }
+}
+
+export function makeModuleMethod(module: QualifiedName, name: string, returnType: string) {
+  return {
+    memberOf: module,
+    ...makeSimpleEntry(SuggestionKind.Method, module, name, returnType),
+  }
+}
+
+export function makeFunction(definedIn: QualifiedName, name: string, returnType: string) {
+  return makeSimpleEntry(SuggestionKind.Function, definedIn, name, returnType)
+}
+
+export function makeLocal(definedIn: QualifiedName, name: string, returnType: string) {
+  return makeSimpleEntry(SuggestionKind.Local, definedIn, name, returnType)
+}
```

Comment:
For functions like those, it's good idea to add an explicit return type. Otherwise some added fields might end up with incorrect type, and it will not be immediately caught.